common/src/main/java/com/wynntils/screens/itemsharing/ItemSharingScreen.java

```diff
@@ -162,17 +184,67 @@ private void refreshPreview() {
     }
 
     private void shareItem(String target) {
-        if (target.equals("guild")) {
-            Handlers.Command.sendCommand("g " + encodedItem.toUtf16String());
-        } else if (target.equals("party")) {
-            Handlers.Command.sendCommand("p " + encodedItem.toUtf16String());
-        } else if (target.equals("clipboard")) {
-            McUtils.mc().keyboardHandler.setClipboard(encodedItem.toUtf16String());
-
-            McUtils.sendMessageToClient(Component.translatable("screens.wynntils.itemSharing.copied")
-                    .withStyle(ChatFormatting.GREEN));
-        } else {
-            // TODO: Save item to storage
+        switch (target) {
+            case "guild" -> Handlers.Command.sendCommand("g " + encodedItem.toUtf16String());
+            case "party" -> Handlers.Command.sendCommand("p " + encodedItem.toUtf16String());
+            case "save" -> {
+                Map<String, Map<String, SavedItemStack>> savedItems = Models.ItemEncoding.savedItems.get();
+                Map<String, SavedItemStack> uncategorisedItems =
+                        savedItems.getOrDefault("Uncategorized", new HashMap<>());
+
+                String encodedBase64 = encodedItem.toBase64String();
+
+                // Check if the item is already saved, loop through categories then check keys (encoded item)
+                for (Map.Entry<String, Map<String, SavedItemStack>> entry : savedItems.entrySet()) {
+                    if (entry.getValue().containsKey(encodedBase64)) {
+                        McUtils.sendMessageToClient(Component.translatable(
+                                        "screens.wynntils.itemSharing.alreadySaved", itemStack.getHoverName())
+                                .withStyle(ChatFormatting.RED));
+                        return;
+                    }
+                }
+
+                ItemStack itemStackToSave = itemStack;
+
+                // Gear items can have their item changed by cosmetics so we need to get their original item
+                // FIXME: Does not work for crafted gear
+                if (wynnItem instanceof GearItem gearItem) {
+                    itemStackToSave = new FakeItemStack(gearItem, "From " + McUtils.playerName() + "'s vault");
+                }
```

Comment:
So what's the problem with crafted gear/consumables?

Reply:
Crafted armour/weapons still show their cosmetics/the shield icon if hidden via silverbull. Everything else works fine

Reply:
Actually, it probably also doesn't work for accessories and consumables as for some reason Wynn changes non armour to the shield item if they're in certain container slots (usually the bottom right corner)
![image](https://github.com/Wynntils/Artemis/assets/8337467/9a35f461-4030-408c-9c91-e22332d13caa)

Reply:
That's acceptable for now.